### PR TITLE
Example

### PR DESCRIPTION
**Pull request by footprinthere**
by *Seongtae Jeong*
Added a **simple MNIST classifier** example and a **Huggingface CLM** example.

### CODE REVIEW

File: examples/huggingface/causal_language_modeling/original/orig_run_clm.py [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+        trainer.create_model_card(**kwargs)
+
+
+def _mp_fn(index):
+    # For xla_spawn (TPUs)
+    main()
+
+
+if __name__ == "__main__":
+    main()
```
Comment (ktaebum) [inline examples/huggingface/causal_language_modeling/original/orig_run_clm.py:563]: Could you add a newline for each file?

File: examples/huggingface/causal_language_modeling/original/pf_config.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# Checkpoint config
+checkpoint:
+  # Path to output checkpoint
+  output_checkpoint_dir: /workspace/ckpt
+
+# Configuration dataset
+data:
+  name: wikitext
+  mount_path: /workspace/data
+
```
Comment (ktaebum) [inline examples/huggingface/causal_language_modeling/original/pf_config.yml:48]: Could you add a newline?

File: examples/huggingface/causal_language_modeling/run_clm.py [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+            if self.validation_file is not None:
+                extension = self.validation_file.split(".")[-1]
+                assert extension in ["csv", "json", "txt"], "`validation_file` should be a csv, a json or a txt file."
+
+
+def print_once(msg):
+    """ Prints a message only in the main process """
+    if torch_ddp.get_rank() != 0:
+        return
+    print(msg, flush=True)
```
Comment (ktaebum) [inline examples/huggingface/causal_language_modeling/run_clm.py:205]: Not using this function?

File: examples/huggingface/causal_language_modeling/run_clm.py [hunk 4]
```diff
@@ -0,0 +1,581 @@
...
+            perplexity = math.exp(metrics["eval_loss"])
+        except OverflowError:
+            perplexity = float("inf")
+        metrics["perplexity"] = perplexity
+
+        trainer.log_metrics("eval", metrics)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (ktaebum) [inline examples/huggingface/causal_language_modeling/run_clm.py:581]: newline is needed

File: examples/mnist/main.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    torch_ddp.init_process_group(backend=backend)
+    world_size = torch_ddp.get_world_size()
+
+    if args.use_cpu:
+        device = None
+        print("Using CPU", flush=True)
+    else:
+        torch.cuda.set_device(int(os.environ["LOCAL_RANK"]))
+        device = torch.cuda.current_device()
+        print_once("Using GPU")
```
Comment (ktaebum) [inline examples/mnist/main.py:58]: remove `print` and `print_once`?

**Comment by footprinthere** [comment]
Made some new modifications according to your review comments.

**Review by ktaebum — APPROVED** [review]

**Pull request merged**